tensorrt_llm/_torch/disaggregation/transceiver.py

```
@@ -350,6 +352,18 @@ def _consensus_outcome(
 
     def _gen_consensus_outcome(self, to_process, cancelled, failed, completed):
+        # adp-off: TP-then-PP two-stage on the ordered subgroup communicators
+        # (not a WORLD allgather) to avoid the cross-communicator PP deadlock;
+        # tp x pp == world, so the outcome is identical.
+        if not self._mapping.enable_attention_dp:
+            c, f, d = self._consensus_outcome(to_process, cancelled, failed,
+                                              completed, self._dist.tp_allgather,
+                                              self._ctx_need_tp_sync)
+            if self._ctx_need_pp_sync:
+                c, f, d = self._consensus_outcome(to_process, c, f, d,
+                                                  getattr(self._dist, "pp_allgather"),
```

Ruff (0.15.18)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)